app/lib/actions/candlepin/consumer/refresh_subscriptions.rb

```diff
@@ -0,0 +1,31 @@
+#
+# Copyright 2014 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+module Actions
+  module Candlepin
+    module Consumer
+      class RefreshSubscrpitions < Actions::EntryAction
+        input_format do
+          param :uuid, String
+        end
+
+        def plan(system, sys_params)
+          plan_self(:uuid => system.uuid) if sys_params[:autoheal]
+        end
+
+        def finalize
+          Resources::Candlepin::Consumer.refresh_entitlements(input[:uuid])
```

Comment:
I think this should be ::Katello::Resources::Candlepin::Consumer  I got a 'uninitialized constant' error.